gwcs/models.py

```diff
@@ -0,0 +1,127 @@
+        The data on a regular grid in n dimensions.
+    points : tuple of ndarray of float, with shapes (m1, ), ..., (mn, ), optional
+        The points defining the regular grid in n dimensions.
+    method : str, optional
+        The method of interpolation to perform. Supported are "linear" and
+        "nearest", and "splinef2d". "splinef2d" is only supported for
+        2-dimensional data. Default is "linear".
+    bounds_error : bool, optional
+        If True, when interpolated values are requested outside of the
+        domain of the input data, a ValueError is raised.
+        If False, then `fill_value` is used.
+    fill_value : float, optional
+        If provided, the value to use for points outside of the
+        interpolation domain. If None, values outside
+        the domain are extrapolated.  Extrapolation is not supported by method
+        "splinef2d".
+
+    Returns
+    -------
+    value : ndarray
+        Interpolated values at input coordinates.
+
+    Raises
+    ------
+    ImportError
+        Scipy is not installed.
+
+    Examples
+    --------
+    >>> from gwcs.models import LookupTable
+    >>> table=np.array([[ 3.,  0.,  0.],
+                        [ 0.,  2.,  0.],
+                        [ 0.,  0.,  0.]])
+    >>> points = [1, 2, 3]
+    >>> xinterp = [0, 1, 1.5, 2.72, 3.14]
+
+    Setting fill_value to None, allows extrapolation.
+
+    >>> ltmodel = LookupTable(table, points, bounds_error=False, fill_value=None, method='nearest')
+    >>> ltmodel(xinterp, xinterp)
+        array([ 3.,  3.,  3.,  0.,  0.])
+
+    Notes
+    -----
+    Uses ``scipy.interpolate.interpn``
+    """
+    linear = False
+    fittable = False
+
+    standard_broadcasting = False
+    outputs = ('lookup_table_value',)
+
+    lookup_table = Parameter()
+
+    def __init__(self, lookup_table, points=None, method='linear',
+                 bounds_error=True, fill_value=np.nan):
+        if not has_scipy:
+            raise ImportError("LookupTable model needs scipy.")
+        if points is None:
+            self._points = tuple(np.arange(x, dtype=np.float) for x in lookup_table.shape)
+        else:
+            self._points = points
+        super(LookupTable, self).__init__(lookup_table=lookup_table)
+        self.inputs = tuple('x' + str(idx)
+                            for idx in range(self.lookup_table.value.ndim))
+
+        self._bounds_error = bounds_error
+        self._method = method
+        self._fill_value = fill_value
+
+    @property
+    def bounds_error(self):
+        return self._bounds_error
+
+    @bounds_error.setter
+    def bounds_error(self, value):
+        self._bounds_error = value
+
+    @property
+    def method(self):
+        return self._method
+
+    @method.setter
+    def method(self, value):
+        self._method = value
+
+    @property
+    def fill_value(self):
+        return self._fill_value
+
+    @fill_value.setter
+    def fill_value(self, value):
+        self._fill_value = value
```

Comment:
Good point. It would work in a multiplicative sense (shouldn't throw any errors would it?), but if adding or whatnot, it would fail. But then again, if parameters don't support it yet, then don't use it until that is ready. It doesn't have to use the parameter machinery though.